From 459d01cbda39e51029c55e583909ef645037b7f3 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Wed, 14 Oct 2015 11:26:27 +0800
Subject: [PATCH] MLK-11700 video: mxsfb: correct the fb pan display ret value

The fb_pan_display() return value should be corrected
according to the wait_for_completion_timeout() return
value.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/video/mxsfb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index f1390168..3fa83db 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -963,13 +963,13 @@ static int mxsfb_pan_display(struct fb_var_screeninfo *var,
 		host->base + LCDC_CTRL1 + REG_SET);
 
 	ret = wait_for_completion_timeout(&host->flip_complete, HZ / 2);
-	if (ret < 0) {
+	if (!ret) {
 		dev_err(fb_info->device,
 			"mxs wait for pan flip timeout\n");
-		ret = -ETIMEDOUT;
+		return -ETIMEDOUT;
 	}
 
-	return ret;
+	return 0;
 }
 
 static int mxsfb_mmap(struct fb_info *info, struct vm_area_struct *vma)
-- 
1.8.0

